modules/util/pathfinder/artifacts/opt/jboss/container/util/pathfinder/pathfinder.sh

```diff
@@ -0,0 +1,105 @@
+# Error is indicated with a prefix in the return value
+check_error() {
+  local msg=$1
+  if echo ${msg} | grep -q "^ERROR:"; then
+    log_error ${msg}
+    exit 1
+  fi
+}
+
+# detect Quarkus fast-jar package type (OPENJDK-631)
+is_quarkus_fast_jar() {
+  if test -f quarkus-app/quarkus-run.jar; then
+    log_info "quarkus fast-jar package type detected"
+    echo quarkus-app/quarkus-run.jar
+    return 0
+  else
+    return 1
+  fi
+}
+
+# Try hard to find a sane default jar-file
+auto_detect_jar_file() {
+  local dir=$1
+
+  # Filter out temporary jars from the shade plugin which start with 'original-'
+  local old_dir=$(pwd)
+  cd ${dir}
+  if [ $? = 0 ]; then
+    if quarkus="$(is_quarkus_fast_jar)"; then
+      echo "$quarkus"
+      return
+    fi
+    local nr_jars=`ls *.jar 2>/dev/null | grep -v '^original-' | wc -l | tr -d '[[:space:]]'`
+    if [ ${nr_jars} = 1 ]; then
+      ls *.jar | grep -v '^original-'
+      exit 0
+    fi
+
+    log_error "Neither \$JAVA_MAIN_CLASS nor \$JAVA_APP_JAR is set and ${nr_jars} JARs found in ${dir} (1 expected)"
+    cd ${old_dir}
+  else
+    log_error "No directory ${dir} found for auto detection"
+  fi
+}
+
+# Check directories (arg 2...n) for a jar file (arg 1)
+get_jar_file() {
+  local jar=$1
+  shift;
+
+  if [ "${jar:0:1}" = "/" ]; then
+    if [ -f "${jar}" ]; then
+      echo "${jar}"
+    else
+      log_error "No such file ${jar}"
+    fi
+  else
+    for dir in $*; do
+      if [ -f "${dir}/$jar" ]; then
+        echo "${dir}/$jar"
+        return
+      fi
+    done
+    log_error "No ${jar} found in $*"
+  fi
+}
+
+setup_java_app_and_lib() {
+  # Configuration stuff is read from this file
+  local run_env_sh="run-env.sh"
+
+  # Load default default config
+  if [ -f "${JBOSS_CONTAINER_JAVA_RUN_MODULE}/${run_env_sh}" ]; then
+    source "${JBOSS_CONTAINER_JAVA_RUN_MODULE}/${run_env_sh}"
+  fi
+
+  # Check also $JAVA_APP_DIR. Overrides other defaults
```

Comment:
This is the code from run-java.sh prior to merging https://github.com/jboss-container-images/openjdk/pull/478 -- please update the pathfinder copy in this PR to reflect the latest logic in the ubi9 branch